daemon/src/monitor.rs

```diff
@@ -298,30 +323,29 @@ where
             self.latest_block_height = latest_block_height;
         }
 
-        // 1. shape response into local data format
-        let new_status = histories.into_iter().zip(self.awaiting_status.keys().cloned()).map(|(script_history, (txid, script))| {
-            let new_script_status = match script_history.as_slice() {
-                [] => ScriptStatus::Unseen,
-                [remaining @ .., last] => {
-                    if !remaining.is_empty() {
-                        tracing::warn!("Found more than a single history entry for script. This is highly unexpected and those history entries will be ignored")
-                    }
-
-                    if last.height <= 0 {
-                        ScriptStatus::InMempool
-                    } else {
-                        ScriptStatus::Confirmed(
-                            Confirmed::from_inclusion_and_latest_block(
-                                u32::try_from(last.height).expect("we checked that height is > 0"),
+        // 1. Decide new status based on script history
+        let new_status = self
+            .awaiting_status
+            .iter()
+            .map(|(key, _old_status)| {
+                let new_script_status = match histories.get(key) {
+                    None => ScriptStatus::Unseen,
+                    Some(history_entry) => {
+                        if history_entry.height <= 0 {
+                            ScriptStatus::InMempool
+                        } else {
+                            ScriptStatus::Confirmed(Confirmed::from_inclusion_and_latest_block(
```

Comment:
As a clean-up, I think it would make sense to get rid of this `Confirmed` struct and just inline the entire logic into `ScriptStatus`.